src/main/gui/main-window.ts

```diff
@@ -573,6 +584,19 @@ export const createMainWindow = async (args: OpenArguments, settings: ChainnerSe
                 path.join(__dirname, `../renderer/${MAIN_WINDOW_VITE_NAME}/index.html`)
             );
         }
+
+        // Scan install dir for chain files and warn user if they are found
+        const chainFiles = await getAllFiles(installDir);
```

Comment:
> Ideally, we'd only scan once after the user updates.

After the user updates, all their files saved here would be gone, so there's no point in even doing this scan in that case. If we really don't want to scan every time, then we should just remove this part. However, since this is done after the main window stuff is set to ready, it shouldn't cause any lag in terms of startup time

Reply:
> After the user updates, all their files saved here would be gone

Then why do we need this in the first place? If the install dir starts out without chain file, and we explicitly prevent the user from saving any chain files in there, then what do we scan for?

Again, if someone manually moves their chain files there, then we can't help them.

Reply:
This is still relevant @joeyballentine, in case you missed it.

Reply:
> Again, if someone manually moves their chain files there, then we can't help them.

The goal was to help out people that do this, because even though its their fault we can still hopefully prevent data loss from their stupid behavior. But i do agree that this is probably unnecessary now that saving directly to the folder is prohibited.

Reply:
we can always add this back if it still ends up being an issue for people after this change